lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -55,85 +57,136 @@
-        if mpi._is_mpirun_for(mpirunname):
-            return scriptname, mpi, found_mpi
+    # return found mpirun_path
+    _logger.warn("The executable that called mympirun isn't supported"
+                 ", defaulting to %s", mpirun_path)
+    return mpirun_path, None, supp_mpi_impl
 
-    # return found mpirunname
-    return mpirunname, None, found_mpi
 
+def stripfake(path_to_append=None):
+    """Removes the fake wrapper path from $PATH
 
-def _setenv(name, value):
-    """Set environment variable. In principle os.environ should be sufficient."""
-    os.putenv(name, "%s" % value)
-    os.environ[name] = "%s" % value
+    assumes (VSC-tools|mympirun)/1.0.0/bin/fake
 
+    Arguments:
+        path_to_append  --  a list of paths that needs to be added to $PATH
 
-def stripfake(path=None):
-    """Remove the fake wrapper path:
-        assumes (VSC-tools|mympirun)/1.0.0/bin/fake
+    Returns:
+        newpath         --  a list of the directories in the new $PATH
+                            (without the fake mpirun and with path_to_append
+                            appended)
     """
-    reg_fakepath = re.compile(r"" + os.sep.join(['.*?', INSTALLATION_SUBDIRECTORY_NAME + '.*?', 'bin',
-                                                '%(fake_subdir)s(%(sep)s[^%(sep)s]*)?$' %
-                                                {
-                                                    'fake_subdir': FAKE_SUBDIRECTORY_NAME,
-                                                    'sep': os.sep
-                                                }]))
-
-    if path is None:
-        path = []
+
+    _logger = getLogger()
+    _logger.debug("mpi.py - PATH before stripfake(): %s", os.environ['PATH'])
+
+    # compile a regex that matches the faked mpirun
+    reg_fakepath = re.compile(
+        r"" + os.sep.join(['.*?',
+                           INSTALLATION_SUBDIRECTORY_NAME + '.*?',
+                           'bin',
+                           '%(fake_subdir)s(%(sep)s[^%(sep)s]*)?$' %
+                           {
+                               'fake_subdir': FAKE_SUBDIRECTORY_NAME,
+                               'sep': os.sep
+                           }
+                           ]))
+
+    # merge the current $PATH and the path_to_append
+    if path_to_append is None:
+        path_to_append = []
     envpath = os.environ.get('PATH', '').split(os.pathsep)
-    # do not append doubles (respect search order)
-    path = envpath + [x for x in path if not x in envpath]
+    newpath = envpath + [x for x in path_to_append if x not in envpath]
 
-    newpath = [x for x in path if not reg_fakepath.match(x)]
+    # remove all $PATH elements that match the fakepath regex
+    newpath = [x for x in newpath if not reg_fakepath.match(x)]
 
-    _setenv('PATH', ':'.join(newpath))
+    os.environ['PATH'] = "%s" % ':'.join(newpath)
 
+    _logger.debug("mpi.py - PATH after stripfake(): %s", os.environ['PATH'])
     return newpath
 
 
 def which(names):
```

Comment:
should I copy paste the function? adding a dependency on easybuild seems inefficient for such a small change